bundles/org.openhab.binding.thekeys/README.md

```diff
@@ -0,0 +1,101 @@
+This binding allows you to integrate, view, control and configure TheKeys Gateway and TheKeys Smartlock.
+
+![The Keys smartlock](doc/thekeys-smartlock.png)
+![The Keys gateway](doc/thekeys-gateway.png)
+
+## Prerequisites
+
+At least one TheKeys Smartlock which is associated with The Keys Gateway.
+The Gateway must be paired with the Smartlock via the android/ios app.
+It is recommended to configure static IP addresses for the Gateway.
+
+### Setup
+
+- Create access for the Gateway from the app or from https://api.the-keys.fr
+- Go to the shares section of the lock (https://api.the-keys.fr/en/compte/serrure)
+- In the accessory shares table, get the code of your Gateway
+
+## Supported Bridges
+
+### `gateway` Bridge Thing Configuration
+
+This binding supports just one bridge type: The Keys Gateway (`thekeys:gateway`).
+The following configuration options are available:
+
+| Name            | Type    | Description                             | Default | Required | Advanced |
+|-----------------|---------|-----------------------------------------|---------|----------|----------|
+| host            | text    | Host or IP address of the gateway       | N/A     | yes      | no       |
+| code            | text    | The gateway access code                 | N/A     | yes      | no       |
+| refreshInterval | integer | Interval the device is polled in sec    | 5       | no       | no       |
+| apiTimeout      | integer | API timeout in seconds for each request | 30      | no       | yes      |
+
+### Discovery
+
+The bridge can't be discovered automatically.
+The `host` parameter must be provided.
+You can find it from the ios/android app : Home Page > Accessories > TK Gateway > Configure > IP
+
+### Supported Channels
+
+The gateway has no channels.
+It only exposes one property `firmwareVersion` containing the firmware version.
+
+## Supported Things
+
+### `smartlock` Thing Configuration
+
+This thing allows to control the smartlock and check its state.
+
+| Name            | Type    | Description                           | Default | Required | Advanced |
+|-----------------|---------|---------------------------------------|---------|----------|----------|
+| lockId          | integer | Identifier of the smartlock           | N/A     | yes      | no       |
+
+### Discovery
+
+The smartlock can be discovered when the Gateway is configured and online.
+Hit the "scan" button when you add the thing through the UI
+
+### Supported Channels
+
+| Channel         | Type         | Read/Write | Description                                    |
+|-----------------|--------------|------------|------------------------------------------------|
+| status          | String       | R          | Status of the smartlock                        |
+| batteryLevel    | Number:Power | R          | Current battery level                          |
+| lowBattery      | Switch       | R          | Low battery warning                            |
```

Comment:
Real channel names are "battery-level" and "low-battery".

Type for battery-level is Number, not Number:Power.